### PR TITLE
add rook ceph docs

### PR DESCRIPTION
**Pull request by Adam-D-Lewis**
## Reference Issues or PRs
Docs going along with Rook Ceph PR - https://github.com/nebari-dev/nebari/pull/2541

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: docs/docs/explanations/advanced-configuration.md [hunk 1]
```diff
@@ -238,20 +237,17 @@ conda_store:
 
 The `nebari-config.yaml` file can be split into several sections.
 
-The first section is the version of Nebari you wish to run.
```
Comment (Adam-D-Lewis) [inline docs/docs/explanations/advanced-configuration.md:241]: This reformatting was done by the pre-commit hooks.

File: docs/docs/explanations/advanced-configuration.md [hunk 2]
```diff
@@ -416,6 +411,62 @@ Defining a wildcard certificate decreases the amount of Common Name (CN) names y
 </TabItem>
 </Tabs>
 
+### Shared Storage Configuration
```
Comment (Adam-D-Lewis) [inline docs/docs/explanations/advanced-configuration.md:414]: This is the only new section.  The rest is just reformatted by precommit.

File: docs/docs/explanations/advanced-configuration.md [hunk 3]
```diff
@@ -416,6 +411,62 @@ Defining a wildcard certificate decreases the amount of Common Name (CN) names y
...
+
+Nebari includes shared file systems for the jupyterhub user storage, jupyterhub shared storage, and conda store shared storage. By default, NFS drives are used.
+
+As of Nebari 2024.9.1, alpha support for [Ceph](https://docs.ceph.com/en/latest/) shared file systems as an alternative to NFS is available. The initial benefit of using Ceph is increased read/write performance compared to NFS, but further benefits are expected in future development. Ceph is a distributed storage system which has the potential to provide increased performance, high availability, data redundancy, storage consolidation, and scalability to Nebari.
+
+:::danger
+Do not switch from one storage type to another on an existing Nebari deployment. Any files in the user home directory and conda environments will be lost if you do so! On GCP, all node groups in the cluster will be destroyed and recreated. Only change the storage type prior to the initial deployment.
+:::
+
+Storage is configuredin the nebari config file under the storage section.
```
Comment (dcmcand) [inline docs/docs/explanations/advanced-configuration.md:424]:
```suggestion
Storage is configured in the nebari config file under the storage section.
```

**Review by dcmcand — CHANGES_REQUESTED** [review]

File: docs/docs/explanations/advanced-configuration.md [hunk 4]
```diff
@@ -416,6 +411,62 @@ Defining a wildcard certificate decreases the amount of Common Name (CN) names y
...
+
+```yaml
+storage:
+  type: nfs  # efs (default on AWS), cephfs
+  conda_store: 200Gi
+  shared_filesystem: 200Gi
+```
+
+If using the "cephfs" storage type. The block storage backing all ceph storage will be provisioned using a kubernetes storage class. The default storage class will be used unless specified. Premium storage can be used on some cloud providers to back ceph storage for improved performance. The storage class name can be specified in the yaml file. Be aware that premium storage is not accessible on all node types on some cloud providers.
+
```
Comment (viniciusdc) [inline docs/docs/explanations/advanced-configuration.md:434]:
```suggestion
When using the `cephfs` storage type option, the block storage underlying all Ceph storage will be provisioned through the same Kubernetes storage class. By default, Kubernetes will use the default storage class unless a specific one is provided. For enhanced performance, some cloud providers offer premium storage class options. 

You can specify the desired storage class under `storage.ceph` section in the configuration file.  Below are examples of potential storage class values for various cloud providers:

```

File: docs/docs/explanations/advanced-configuration.md [hunk 5]
```diff
@@ -416,6 +411,62 @@ Defining a wildcard certificate decreases the amount of Common Name (CN) names y
...
+  storage_class_name: some-cluster-storage-class
+```
+
+  </TabItem>
+  <TabItem label="Local" value="Local">
+
+Ceph is not supported on local deployments.
+</TabItem>
+</Tabs>
+
```
Comment (viniciusdc) [inline docs/docs/explanations/advanced-configuration.md:469]:
```suggestion
:::note
Premium storage might not be available for certain cloud providers on all node types, which can affect performance and availability. Make sure to check the documentation for your specific cloud provider to confirm premium storage options and their compatibility with your node types.
:::
```

File: docs/docs/explanations/advanced-configuration.md [hunk 6]
```diff
@@ -416,6 +411,62 @@ Defining a wildcard certificate decreases the amount of Common Name (CN) names y
...
+<Tabs>
+  <TabItem label="AWS" value="AWS" default="true">
+
+Premium storage is not available on AWS.
+</TabItem>
+<TabItem label="Azure" value="Azure">
+
+```yaml
+ceph:
+  storage_class_name: managed-premium
```
Comment (viniciusdc) [inline docs/docs/explanations/advanced-configuration.md:444]:
```suggestion
storage:
  ...
  ceph:
    storage_class_name: managed-premium
```
This was not clear to me in the first glance, might be worth updating the other examples

Comment (Adam-D-Lewis) [reply docs/docs/explanations/advanced-configuration.md:449]: The change you've made doesn't work.  ceph is a top level key in nebari config file.

Comment (Adam-D-Lewis) [reply docs/docs/explanations/advanced-configuration.md:452]: I see that I had a wrong explanation.  I fixed it now.

File: docs/docs/explanations/advanced-configuration.md [hunk 7]
```diff
@@ -416,6 +411,62 @@ Defining a wildcard certificate decreases the amount of Common Name (CN) names y
 </TabItem>
 </Tabs>
 
+### Shared Storage Configuration
+
```
[the remark this answers is not in the document]
Comment (Adam-D-Lewis) [reply docs/docs/explanations/advanced-configuration.md:415]: accepted

**Review by viniciusdc — APPROVED** [review]
LGTM, thanks @Adam-D-Lewis

**Pull request merged**